includes/functions_install.php

```diff
@@ -136,10 +136,20 @@ function load_schema_31($install_path = '', $install_dbms = false)
 			include($phpbb_root_path . 'includes/constants.' . $phpEx);
 		}
 
-		$finder = new \phpbb\finder(new \phpbb\filesystem(), $phpbb_root_path, null, $phpEx);
+		// Check if equal or greater than 4.0@dev
+		if (file_exists($phpbb_root_path . 'vendor-ext'))
```

Comment:
Instead of this, define a PHPBB_40 constant in the index, like the way we defined PHPBB_31 and PHPBB_32. Then check here if PHPBB_40 is defined

Reply:
I kind of want to avoid that as it's kind of a bad practice to be honest.

Reply:
But that's literally how all of QuickInstall is managing all the different coding requirements between the multiple phpBB versions. We can't have exotic tests everywhere, instead we use a defined constant that can be used everywhere checks are needed. 

Also it's for consistency and continuity as well. If it's a bad practice, it's too late as it's already everywhere in here (and most of QI code is bad practice too IMO, like using phpBB 3.0's old template engine and syntax, but that's another issue).

Reply:
I'm aware of how the rest of the app is written and that there's this nonsense all over it. This is fine for now I guess.

Reply:
I agree with you fully on the nonsense 😃

Reply:
The use of constants has been discussed within this PR
https://github.com/phpbb/quickinstall/pull/112
where I started following the "logic" of the actual code, for consistency.

At some point this app should be rewritten IMO, so I agree with David.

Reply:
Yes, and it should continue to be followed here. It's the only way to find where we're making special accomodations for specific phpBB versions... by searching for where these constants are being used.

You can not simply start creating new tests based on arbitrary conditions to accomodate phpBB specific code. That makes it VERY difficult to find phpBB version specific code in this application.

Reply:
Of course, for better or for worse, we're all right, it seems, and I agree. I would like to know exactly what approach to follow then. Obviously I consider the use of all these constants a little illogical, we can argue that even the present method is otherwise illogical.

It is up to you, then, to choose.

Reply:
constants

Reply:
Okay, I will push the changes so we can see if the new approach satisfies we all.

Reply:
Just to expand on that: If anybody wants to come up with an alternative to these constants that actually works well, that's great but that's also a separate project for another PR. For this PR, and for the time being, we should continue handling these different version-specific code conditions the way the app currently does it, which is by these constants.